File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    const stack = []


    tokens.forEach((token) => {
        if (!isNaN(token)) {
            stack.push(Number(token))
        } else {
            const b = stack.pop()
            const a = stack.pop()

            switch (token) {
                case '+':
                    stack.push(a + b)
                    break
                case '-':
                    stack.push(a - b)
                    break
                case '*':
                    stack.push(a * b)
                    break
                case '/':
                    stack.push(a / b > 0 ? Math.floor(a / b ) : Math.ceil(a / b))
            }
        }
    })
    return stack.pop()
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        ops = {
            '+': operator.add,
            '-': operator.sub,
            '*': operator.mul,
            '/': lambda a,b: int(a/b)
        }

        for t in tokens:
            if t in ops:
                b, a = stack.pop(), stack.pop()
                result = ops[t](a, b)
                stack.append(result)
            else:
                stack.append(int(t))
        
        return stack[0]
            